api/envoy/config/core/v3/protocol.proto

```diff
@@ -313,10 +316,13 @@ message Http2ProtocolOptions {
   // of WINDOW_UPDATE frames received over the lifetime of connection exceeds the value calculated
   // using this formula::
   //
-  //     1 + 2 * (inbound_streams +
-  //              max_inbound_window_update_frames_per_data_frame_sent * outbound_data_frames)
+  //   max_inbound_window_update_frames_per_data_frame_sent * (outbound_data_frames + opened_streams + 1)
```

Comment:
The reasoning is to make tunable allowed number of WINDOW_UPDATE frames **before** any DATA was sent. This makes cases like facebook's or cases where server may preemptively open stream window knowing that it has large body work better.
I think agree that 2x multiplier needs to be put back.
Note that in the more restrictive case where `max_inbound_window_update_frames_per_data_frame_sent == 1` the new formula (with 2x multiplier put back) does not increase the limit compared to the old formula.

Reply:
> The reasoning is to make tunable allowed number of WINDOW_UPDATE frames before any DATA was sent. This makes cases like facebook's or cases where server may preemptively open stream window knowing that it has large body work better.

My point is that if there is a specific scenario that we want to address (e.g. upstream servers sending 2 `WINDOW_UPDATE` frames for each new stream), then it should be part of the formula, and we could either hardcode the value we deem reasonable (e.g. `2 * inbound_streams`) or introduce a new tunable. This PR should improve the formula, and not simply increase the limit.

Also, using `max_inbound_window_update_frames_per_data_frame_sent` to control number of `WINDOW_UPDATE` frames allowed before receiving any `DATA` frames seems quite confusing.

Reply:
Adding another variable increases complexity of the configuration. However I do not think it adds much value. This limit was put in place to address dribble of a large response with 1 byte windows updates. This value does not need to provide exact limit on the number of WINDOW_UPDATE frames a peer may send, just prevent abuse.
It does increase the limit for the default value of 10 which was too large anyway. However for the recommended `max_inbound_window_update_frames_per_data_frame_sent` value of 1 it increases the limit by 1 to accommodate additional initial connection WINDOW_UPDATE. It also addresses the case that @antoniovicente brought up.
The new formula also allows you to effectively remove the limit by setting `max_inbound_window_update_frames_per_data_frame_sent` to UINT32_MAX.
I think this can be considered an improvement to the formula as it solves all of the problems I've set out to solve without adding extra config knob.
If the naming of the tunable multiplier is confusing we can rename it to something like `max_window_update_multiplier` to make it not being tied to DATA frames specifically.
What do you think?

Reply:
I'm going to just add a fixed constant to solve the issue we have in prod. We can discuss the formula later and do the deprecation dance if we need to.